### ISSUE TITLE
LTX-2.5 on Apple Silicon (MPS) generates completely green video on LTX Desktop 1.2.7

### ISSUE THREAD

**Issue by Hpdjoy** [Lightricks/LTX-Desktop#165]
<img width="1416" height="918" alt="Image" src="https://github.com/user-attachments/assets/76b2abf7-fbe0-4ad5-ad75-1eba122f3cc5" />
Environment

LTX Desktop: 1.2.7
OS: macOS Sequoia 15.3.1
Hardware: Mac Studio, Apple M1 Ultra, 128 GB Unified Memory
PyTorch: 2.11.0
mps_sdpa: 0.2.0
Generation: Local
Model: LTX-2.5 BF16

Issue

When generating a video locally using LTX Desktop 1.2.7 on Apple Silicon, the generated video is completely green.

This happens with both:

Text-to-Video (T2V)
Image-to-Video (I2V)

The output MP4 is successfully created and saved, but all frames appear as a solid green image.

What I observed

The model loads successfully and the denoising process completes without an apparent runtime failure.

The logs show:

Building transformer with attention backends -- self: MPS-SDPA, masked: MPS-SDPA

The denoising loop also completes successfully, for example:

Running denoising loop (8 steps, 512x288 121 frames @24 fps)

The video VAE/audio VAE are subsequently loaded and the MP4 is successfully written.

There is no obvious:

Traceback
RuntimeError
Exception

in the generation log.

I also tested MPS attention separately:

PyTorch MPS FP32 attention works
PyTorch MPS BF16 attention works
mps_sdpa.sdpa_opt() BF16 test works
No NaN/Inf was observed in these standalone tests

System memory usage also remained below the available 128 GB unified memory.

Suspected cause

The issue appears to be related to the LTX-2.5 Apple/MPS execution path, particularly the custom MPS-SDPA attention backend, rather than a model-loading or memory issue.

LTX's attention implementation automatically selects MPS-SDPA on macOS, and the transformer is reported as using:

self: MPS-SDPA
masked: MPS-SDPA

I have not yet confirmed that this is the root cause, but forcing the transformer to use the standard PyTorch attention implementation may be a useful diagnostic.

Expected behavior

The generated video should contain the requested scene/content normally.

Actual behavior

The generated MP4 is completely green.

Question

Could the maintainers confirm whether LTX-2.5/LTX Desktop 1.2.7 is currently supported on Apple Silicon/MPS for local generation?

Also, is there a recommended way to force the LTX transformer to use PyTorch SDPA / standard MPS attention instead of MPS-SDPA for testing?

I can provide the complete generation logs and additional diagnostic information if required.

**Comment by ericlsimplifi** [comment]
Having the exact same issue

**Comment by ruramire** [comment]
I was able to isolate the green-video issue on macOS / Apple Silicon with LTX Desktop 1.2.7.

LTX 2.5 itself works correctly on MPS: direct generation with the same models, Torch 2.11, BF16, mpsgraph_zc, ConvVAE and OffloadMode.DISK produces normal video and audio.

The problem appears to come from the global patch installed by LTX Desktop:

```python
import services.patches.mps_sdpa_torch as _mps_sdpa_torch
_mps_sdpa_torch.install()
``` 

This globally overrides torch.nn.functional.scaled_dot_product_attention on MPS. LTX Video already uses its own MPS-SDPA backend, and this extra global hook seems to interfere with another part of the pipeline, resulting in a corrupted final latent and a completely green video.

A working fix on macOS is simply to avoid installing this global patch:

```python
if sys.platform != "darwin":
    _mps_sdpa_torch.install()
``` 

With that change, LTX 2.5 generates both video and audio correctly, while LTX’s own MPS-SDPA remains enabled (self: MPS-SDPA, masked: MPS-SDPA).

I also tested and ruled out the VAE, BF16, MP4 encoding, mpsgraph_zc, safetensors loader/metadata patches, diffvae_decode_vram, and torch.inference_mode() as the cause.
